pkgs/os-specific/linux/audit/default.nix

```diff
@@ -55,6 +56,13 @@ stdenv.mkDerivation rec {
 
   prePatch = ''
     sed -i 's,#include <sys/poll.h>,#include <poll.h>\n#include <limits.h>,' audisp/audispd.c
+  ''
+  # According to https://stackoverflow.com/questions/13089166
+  # --whole-archive linker flag is required to be sure that linker
+  # correctly chooses strong version of symbol regardless of order of
+  # object files at command line.
+  + stdenv.lib.optionalString stdenv.targetPlatform.isStatic ''
```

Comment:
This should have been `hostPlatform`

Reply:
Ah, done in 0034539abab37c4cfbc281a084cf111a16e8a21d thanks @malbarbo! Just need to backport to 21.05.